custom_components/tuya_local/devices/kogan_lx10_vacuum.yaml

```diff
@@ -0,0 +1,259 @@
+        - dps_val: backcharge
+          value: backcharge
+    - id: 105
+      name: robot_state
+      type: string
+      mapping:
+        - dps_val: idle
+          value: idle
+        - dps_val: pointing\areaing
+          value: pointing\areaing
+        - dps_val: totaling
+          value: totaling
+        - dps_val: sweep
+          value: sweep
+        - dps_val: mop
+          value: mop
+        - dps_val: fault
+          value: fault
+        - dps_val: pause
+          value: pause
+        - dps_val: chargring
+          value: chargring
+        - dps_val: tocharge
+          value: tocharge
+        - dps_val: fullcharge
+          value: fullcharge
+        - dps_val: remotectl
+          value: remotectl
+    - id: 106
+      name: battery
+      type: integer
+    - id: 107
+      name: cur_clean_time
+      type: integer
+    - id: 108
+      name: cur_clean_area
+      type: integer
+    - id: 109
+      name: fan_mode
+      type: string
+      mapping:
+        - dps_val: quiet
+          value: quiet
+        - dps_val: strong
+          value: strong
+        - dps_val: auto
+          value: auto
+    - id: 110
+      name: water_mode
+      type: string
+      mapping:
+        - dps_val: 'off'
+          value: 'off'
+        - dps_val: quiet
+          value: quiet
+        - dps_val: auto
+          value: auto
+        - dps_val: strong
+          value: strong
+    - id: 111
+      name: remote_ctrl
+      optional: true
+      type: string
+      mapping:
+        - dps_val: forward
+          value: forward
+        - dps_val: backward
+          value: backward
+        - dps_val: left
+          value: left
+        - dps_val: right
+          value: right
+        - dps_val: stop
+          value: stop
+    - id: 112
+      name: seek_robot
+      optional: true
+      type: boolean
+    - id: 113
+      name: disturb_switch_1
+      type: boolean
+    - id: 114
+      name: volume
+      type: integer
+      range:
+        min: 0
+        max: 10
+    - id: 115
+      name: material_reset
+      optional: true
+      type: string
+      mapping:
+        - dps_val: gettime
+          value: gettime
+        - dps_val: resetsidebrush
+          value: resetsidebrush
+        - dps_val: resetmainbrush
+          value: resetmainbrush
+        - dps_val: resetfilter
+          value: resetfilter
+    - id: 116
+      name: total_clean_time
+      type: integer
+      unit: min
+    - id: 117
+      name: total_clean_area
+      type: integer
+      unit: m²
+    - id: 118
+      name: total_clean_count
+      type: integer
+    - id: 119
+      name: side_brush_time
+      type: integer
+      unit: s
+    - id: 120
+      name: main_brush_time
+      type: integer
+      unit: s
+    - id: 121
+      name: filter_time
+      type: integer
+      unit: s
+    - id: 122
+      name: robot_fault
+      # TODO: correctly parse the bitfield
+      type: boolean # should result in non-zero showing as true
```

Comment:
Rename this to `error` to have it affect the status of the vacuum.  Type must match the underlying type, so integer or bitfield, not boolean.
If you identify the individual bits, you can create additional binary_sensors for specific errors or groups of errors under secondary_entities, and/or map the values to meaningful strings for the error attribute here.

Reply:
The mappings here are taken from another integration - I have no idea how the numbers map to actual fault statuses.